unity-client/Assets/Scripts/MainScripts/DCL/Components/QuestTrackingInfo/QuestTrackingInfo.cs

```diff
@@ -0,0 +1,44 @@
+﻿using DCL.Components;
+using DCL.QuestsController;
+using System.Collections;
+using UnityEngine;
+
+public class QuestTrackingInfo : BaseComponent
+{
+    private QuestModel model;
+    private static IQuestsController questsController => QuestsController.i;
+
+    public override object GetModel()
+    {
+        return model;
+    }
+
+    public override void SetModel(object newModel)
+    {
+        if(!(newModel is QuestModel quest))
+            return;
+
+        if (model != null)
+            questsController.RemoveQuest(model);
+
+        model = quest;
+        if(model != null)
+            questsController.UpdateQuestProgress(quest);
+    }
+
+    public override IEnumerator ApplyChanges(string newJson)
+    {
+        if (newJson == "{}")
+        {
+            ApplyChanges(null);
```

Comment:
This shouldn't make the check redundant?

1. The first time we get here with `{}`
2. Second time we pass over the early exit and call `SetModel` anyways.

Reply:
Oh geez... I had an `ApplyChanges(QuestModel quest)` before that was replaced by the new `SetModel` method that arrived from `master`. Really nice catch!